examples/nlp/pretraining_BERT.py

```diff
@@ -0,0 +1,513 @@
+predict if the second sentence in the pair is the subsequent sentence in the original
+document. During training, 50% of the inputs are a pair in which the second sentence is the
+subsequent sentence in the original document, while in the other 50% a random sentence
+from the corpus is chosen as the second sentence. The assumption is that the random sentence
+will be disconnected from the first sentence.
+
+Though Google provides a pre-trained checkpoint of BERT on English, you might often need
+to either pre-train the model from scratch for a different language, or do a
+continued-pretraining to fit the model to your domain. In this notebook, we will pre-train
+BERT from scratch optimizing both MLM and NSP objectves using 🤗 Transformers on the `WikiText`
+English dataset loaded from 🤗 Datasets.
+"""
+
+"""
+## Setup
+"""
+
+"""
+### Installing the requirements
+"""
+
+"""shell
+pip install git+https://github.com/huggingface/transformers.git
+pip install datasets
+pip install huggingface-hub
+pip install nltk
+"""
+
+"""
+### Importing the necessary libraries
+"""
+
+import nltk
+import random
+import logging
+
+import tensorflow as tf
+from tensorflow import keras
+
+nltk.download("punkt")
+# Only log error messages
+tf.get_logger().setLevel(logging.ERROR)
+# Set random seed
+tf.keras.utils.set_random_seed(42)
+
+"""
+### Define certain variables
+"""
+
+TOKENIZER_BATCH_SIZE = 256  # Batch-size to train the tokenizer on
+TOKENIZER_VOCABULARY = 25000  # Total number of unique subwords the tokenizer can have
+
+BLOCK_SIZE = 128  # Maximum number of tokens in an input sample
+NSP_PROB = 0.50  # Probability that the next sentence is the actual next sentence in NSP
+SHORT_SEQ_PROB = 0.1  # Probability of generating shorter sequences to minimize the mismatch between pre-training and fine-tuning.
+MAX_LENGTH = 512  # Maximum number of tokens in an input sample after padding
+
+MLM_PROB = 0.2  # Probability with which tokens are masked in MLM
+
+TRAIN_BATCH_SIZE = 2  # Batch-size for pre-training the model on
+MAX_EPOCHS = 1  # Maximum number of epochs to train the model for
+LEARNING_RATE = 1e-4  # Learning rate for training the model
+
+MODEL_CHECKPOINT = "bert-base-cased"  # Name of pre-trained model from 🤗 Model Hub
+
+"""
+## Load the WikiText dataset
+"""
+
+"""
+We will now download the WikiText language modeling dataset. It is a collection of over
+100 million tokens extracted from the set of verified Good and Featured articles on
+Wikipedia.
+
+We will load the dataset from [🤗 Datasets](https://github.com/huggingface/datasets).
+For the purpose of demonstration in this notebook, we will work with only the `train`
+split of the dataset. This can be easily done with the `load_dataset` function.
+"""
+
+from datasets import load_dataset
+
+dataset = load_dataset("wikitext", "wikitext-2-raw-v1")
+
+"""
+The dataset just has one column which is the raw text, and this is all we need for
+pre-training BERT!
+"""
+
+dataset
```

Comment:
The file should serve as a standalone Python script, so anything you want to display should use `print()`. Don't assume notebook behavior